### DISCUSSION TITLE
Searching for sentences doesn't work

### DISCUSSION THREAD

**Discussion by Pookeyblow** [scambier/obsidian-omnisearch#387] [Q&A]
When I search for 1 word or 2 words, this plugin works great. It highlights the correct text in search just like I want it to.

But when I try to search for longer sentences, it doesn't work at all. Even when I add quotation marks around the sentence, it still doesn't find it. What is wrong? Do I need to change a setting?

What I've noticed by searching various sentences from my vault is that it will often show the right note, but highlight completely different words than those I searched for.

See the screenshots below. Why does it highlight words like "are", "as", "aesthetically", "isolation" and "and" inside the word "Brand"? This is the last thing I want it to do. I searched for none of those words!

This makes no sense to me. I just want to find the sentence "Why is this a good time...", which is right there in the current document I'm viewing. 

![omnisearch1](https://github.com/user-attachments/assets/12fea8aa-ca6c-4157-84b4-b6850bcb8542)
![omnisearch2](https://github.com/user-attachments/assets/26f2f91b-bd87-4a76-93dc-7bdcd6c26f90)

**Comment by scambier** [comment]
> Searching for sentences doesn't work

Yup. Omnisearch is not made to find exact sentences (the core search plugin is much better for this), **it's made to find notes from relevant keywords**. See [this page](https://publish.obsidian.md/omnisearch/How+to+use+Omnisearch) on the documentation site.

When you search for a `"text in quotes"`, Omnisearch will still look for notes that match `text`, `in`, and `quotes`, _and then_ will refine those results to only keep the ones that have the exact string "text in quotes".

> Why does it highlight words like "are", "as", "aesthetically", (...) and "and" inside the word "Brand"? This is the last thing I want it to do. I searched for none of those words!

Omnisearch will try to match your search terms as prefixes. E.g. "for" will also match "force" and "fork". Since you're searching for "a", there are many words that begin with that. It's also why the `and` in "Branding" is highlighted. (The highlighting algorithm is a bit buggy but mostly works fine.)

Hope that answers your questions :)

**Reply by Pookeyblow** [reply]
I understand, but if a sentence matches my search query, why not include it directly in the search results? At the very least, provide a toggleable option to display sentence matches.

It would make the search experience much better. 

If I remember a specific sentence from a document and the search tool can't find it, it's difficult to consider it effective. And the need to switch between the nice spotlight-like search of Omnisearch and the tiny sidebar search is frustrating.

> When you search for a "text in quotes", Omnisearch will still look for notes that match text, in, and quotes, and then will refine those results to only keep the ones that have the exact string "text in quotes".

That makes sense, but showing results for single letters is unhelpful. I searched for "why is this a good" and got irrelevant results including words that started or contained the letter "a", instead of the actual sentence I was looking for. 

How is getting a result for "str**a**tegy" helpful when my search was "why is this a good"?

**Reply by scambier** [reply]
I can succinctly answer with "yes it's possible, but that needs work".

So either when I got enough time and motivation to work on this and one of the 43 pending issues, or if someone else is willing to do it :p

**Comment by Pookeyblow** [comment]
Also, I hope I didn’t come across as too critical. I love the plug-in and use it all the time! I just thought I’d mention something that could make it even better :)
